### PR TITLE
feat(err): Rewrite main, improve error handling, more metrics

### PR DESCRIPTION
**Pull request by oliverb123**
Does what it says on the tin. I want to know more about how long various stages of symbol set processing take, and I want to be able to run the code locally under something like `perf` to profile it and work on hotspots. If we deploy this, I think it'll make cymbal restart much more often, since I've removed the crutch of reporting liveness every frame resolve, but I'm not certain. I kind of suspect the occasional restarts were due to our `reqwest` client not having a low enough timeout, but these metrics should help me say for sure.

Basically, cymbal lives, now lets make it more fragile so we can then make it more robust

### CODE REVIEW

File: rust/cymbal/src/fingerprinting.rs [hunk 1]
```diff
@@ -16,83 +16,11 @@ pub fn generate_fingerprint(exception: &[Exception]) -> String {
 
 #[cfg(test)]
 mod test {
+
     use crate::{frames::Frame, types::Stacktrace};
 
     use super::*;
 
-    #[test]
-    fn test_fingerprint_generation() {
```
Comment (daibhin) [inline rust/cymbal/src/fingerprinting.rs:24]: Why are we removing this test?

File: rust/cymbal/src/frames/mod.rs [hunk 2]
```diff
@@ -17,11 +18,18 @@ pub enum RawFrame {
...
-    pub async fn resolve(&self, team_id: i32, catalog: &Catalog) -> Result<Frame, Error> {
+    pub async fn resolve(&self, team_id: i32, catalog: &Catalog) -> Result<Frame, UnhandledError> {
         let RawFrame::JavaScript(raw) = self;
 
         let frame_resolve_time = common_metrics::timing_guard(PER_FRAME_TIME, &[]);
         let res = raw.resolve(team_id, catalog).await;
+
+        // The raw id of the frame is set after it's resolved
+        let res = res.map(|mut f| {
+            f.raw_id = self.frame_id();
```
Comment (daibhin) [inline rust/cymbal/src/frames/mod.rs:29]: Nice, thanks!

File: rust/cymbal/src/main.rs [hunk 3]
```diff
@@ -51,12 +49,12 @@ fn start_health_liveness_server(config: &Config, context: Arc<AppContext>) -> Jo
...
 #[tokio::main]
-async fn main() -> Result<(), Error> {
+async fn main() {
     setup_tracing();
     info!("Starting up...");
 
-    let config = Config::init_from_env()?;
-    let context = Arc::new(AppContext::new(&config).await?);
+    let config = Config::init_from_env().unwrap();
+    let context = Arc::new(AppContext::new(&config).await.unwrap());
```
Comment (daibhin) [inline rust/cymbal/src/main.rs:57]: I'm guessing unwraps are preferred here because it is basically impossible?

File: rust/cymbal/src/lib.rs [hunk 4]
```diff
@@ -7,3 +18,130 @@ pub mod langs;
...
+
+    props.fingerprint = Some(generate_fingerprint(&results));
+    props.exception_list = Some(results);
+
+    // "But we can open the box" said toad.
+    let mut event = event;
+    event.properties = Some(serde_json::to_string(&props).unwrap());
+
+    // We've processed the exception event succesffully, so we can store the offset.
+    offset.store().unwrap();
```
Comment (daibhin) [inline rust/cymbal/src/lib.rs:63]: Do we want to be storing the offset before we've written to kafka?

**Review by daibhin — APPROVED** [review]

File: rust/cymbal/src/main.rs (hunk 3 above)
Comment (oliverb123) [reply rust/cymbal/src/main.rs:57]: Basically, I'd rather just fail to start up than let callers try and handle this error (it just gets returner by main anyway, which then panics).

File: rust/cymbal/src/lib.rs (hunk 4 above)
Comment (oliverb123) [reply rust/cymbal/src/lib.rs:63]: Yes, I think so - once we have kafka emitting we'll probably do that here too, or maybe pass the final event and the offset along to some buffered flush
